sonar-plugin-api/src/main/java/org/sonar/api/server/rule/internal/DefaultNewRule.java

```diff
@@ -84,6 +85,7 @@ class DefaultNewRule extends RulesDefinition.NewRule {
   private RuleScope scope;
   private final Set<RuleKey> deprecatedRuleKeys = new TreeSet<>();
   private final List<RuleDescriptionSection> ruleDescriptionSections = new ArrayList<>();
+  private final Set<String> genericConceptKeys = new TreeSet<>();
```

Comment:
why not a hashset?

Reply:
I used the Treeset as it is always used for the other string collections in the API, I like the idea of keeping a consistent order for the entries for the definition. (And it also implicitily adds order consistency for the tests)

Reply:
Hmmm ok, I feel it's a bit weird to use a TreeSet here as what we need is to store generic concept keys and order is not defined (alphabetical order is necessarily the right one).

As it is not exposed, nor documented by the API and as it will be more consistent with other fields in that class it's ok for me to leave it as is.